src/plugins/intel_gpu/src/graph/graph_optimizer/add_required_reorders.cpp

```diff
@@ -197,8 +197,13 @@ void add_required_reorders::run(program& p) {
 
             if (!correct_layout_selected) {
                 for (auto new_layout_format : preferred_layout_formats) {
-                    layout current_layout(original_layout.data_type, new_layout_format, original_layout.get_tensor());
-                    usr->set_output_layout(current_layout, false);
+                    if (p.get_options().get<build_option_type::allow_new_shape_infer>()->enabled()) {
+                        layout current_layout(original_layout.get_partial_shape(), original_layout.data_type, new_layout_format);
```

Comment:
Can't it be used for legacy shape infer as well? Do we really need to separate new/old shape infer here?

Reply:
I think its no problem. I will update and check the impact on CI / benchmark

Reply:
Updated